databuilder/extractor/dashboard/tableau/tableau_dashboard_extractor.py

```diff
@@ -0,0 +1,139 @@
+        Dashboard name (Workbook name)
+        Dashboard description (Workbook description)
+        Dashboard creation timestamp (Workbook creationstamp)
+        Dashboard group name (Workbook top-level folder name)
+    As with all the Tableau extractors, uses the Metadata API: https://help.tableau.com/current/api/metadata_api/en-us/index.html
+    """
+
+    API_VERSION = const.API_VERSION
+    TABLEAU_HOST = const.TABLEAU_HOST
+    SITE_NAME = const.SITE_NAME
+    TABLEAU_ACCESS_TOKEN_NAME = const.TABLEAU_ACCESS_TOKEN_NAME
+    TABLEAU_ACCESS_TOKEN_SECRET = const.TABLEAU_ACCESS_TOKEN_SECRET
+    EXCLUDED_PROJECTS = const.EXCLUDED_PROJECTS
+    EXTERNAL_CLUSTER_NAME = const.EXTERNAL_CLUSTER_NAME
+    EXTERNAL_SCHEMA_NAME = const.EXTERNAL_SCHEMA_NAME
+    EXTERNAL_TABLE_TYPES = const.EXTERNAL_TABLE_TYPES
+    CLUSTER = const.CLUSTER
+    DATABASE = const.DATABASE
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+
+        self._conf = conf
+        self._auth = TableauDashboardAuth(self._conf)
+        self.query = """query {
+            workbooks {
+                id
+                name
+                createdAt
+                description
+                projectName
+                projectVizportalUrlId
+                vizportalUrlId
+            }
+        }"""
+
+        self._extractor = self._build_extractor()
+
+        transformers = []
+        timestamp_str_to_epoch_transformer = TimestampStringToEpoch()
+        timestamp_str_to_epoch_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, timestamp_str_to_epoch_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict({FIELD_NAME: 'created_timestamp', })))
+        transformers.append(timestamp_str_to_epoch_transformer)
+
+        dict_to_model_transformer = DictToModel()
+        dict_to_model_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, dict_to_model_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict(
+                    {MODEL_CLASS: 'databuilder.models.dashboard.dashboard_metadata.DashboardMetadata'})))
+        transformers.append(dict_to_model_transformer)
+        self._transformer = ChainedTransformer(transformers=transformers)
+
+    def extract(self):
+        # type: () -> Any
+
+        record = self._extractor.extract()
+        if not record:
+            return None
+
+        return self._transformer.transform(record=record)
+
+    def get_scope(self):
+        # type: () -> str
+
+        return 'extractor.tableau_dashboard_metadata'
+
+    def _build_extractor(self):
+        """
+        Builds a TableauGraphQLApiMetadataExtractor. All data required can be retrieved with a single GraphQL call.
+        :return: A TableauGraphQLApiMetadataExtractor that provides core dashboard metadata.
+        """
+        # type: () -> TableauGraphQLApiMetadataExtractor
```

Comment:
Type hint before docstring
```suggestion
        # type: () -> TableauGraphQLApiMetadataExtractor
        """
        Builds a TableauGraphQLApiMetadataExtractor. All data required can be retrieved with a single GraphQL call.
        :return: A TableauGraphQLApiMetadataExtractor that provides core dashboard metadata.
        """
```

Reply:
This type hinting stuff will go away if/when the true typing PR comes through.